### DISCUSSION TITLE
need help on mobile's client which takes and processes photo

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2718] [Q&A]
### Question

I have read https://github.com/zauberzeug/NiceGUI/discussions/2028, which looks nice.

I have a similar case, but mobile phone clients is desired. You can image the situation that store-keeper meets in his small shop where he has to record goods info on every shelves.
This is my initial thoughts, I marked the steps where I don't know how to implement. Any sugguestion or help? Thanks

| need helps on                                            | description                                                  |
| -------------------------------------------------------- | ------------------------------------------------------------ |
|                                                          | 1. run a NiceGUI server                                      |
|                                                          | 2. on his app/WebApp in mobile phone, store-keeper set the shelf info, for example `Shelf12-Layer1` |
| transferring photo taken by mobile to the NiceGUI server | 3. store-keeper take photos of label on goods then transfer the photos to the NiceGUI server |
| interactive selection useful text area in mobile         | 4. NiceGUI server recognize the strings and barcode on photos, then feed the decoded info back to mobile phone client. In my case, barcode on goods has limited info(i.e. an ID in my case, for example ID20240319001), so extra info(goods name, for example, HP 1020 printer) are supplied by strings printed on label |
|                                                          | 5. store-keeper checks the decoded info, and corrects the wrong characters on his phone. This is necessary because we know OCR can not make absolutely right answer every time |
|                                                          | 6. once the info is confirmed, store-keeper can save entire record(Shelf12-Layer1, ID20240319001, HP 1020 printer) |
|                                                          | 7. if all goods on same shelf have been recorded, go to `step 2`; else go to `step 3` |

**Comment by rodja** [comment]
Sorry. Your question is very vague and we like to have separate questions for different problems. Could you please decouple the topics and state what you have tried already?
